plugins/rapture/nexus-coreui_legacy-plugin/src/main/resources/static/rapture/NX/coreui_legacy/controller/Repositories.js

```diff
@@ -112,7 +112,7 @@ Ext.define('NX.coreui_legacy.controller.Repositories', {
           beforedestroy: me.stopStatusPolling
         },
         'nx-coreui_legacy-repository-feature button[action=browse]': {
-          click: me.navigateToBrowseMode
+          runaction: me.navigateToBrowseMode
```

Comment:
Because I trigger navigation from the “handler” config option, which is in the view, while the code to perform mode switching is in the controller. I use the “runaction” event to pass control from the view to the controller.

Why don’t I just override the “click” handler? When a form is dirty, I cancel navigation. I do this by returning “false” from the click event handler (which prevents the “handler” function, and–by proxy–the “truncation” event). If I used “click” for both dirty checking and navigation, this would not work.

And yeah, it’s really fugly. Any suggestions?